author John Dey Fred Hutch
fix Makefile to use standard GNU Make variable CC, CXX, CFLAGS, CPPFLAGS, CXXFLAGS
diff -ruN MuSE-2.0.1.orig/Makefile MuSE-2.0.1/Makefile
--- MuSE-2.0.1.orig/Makefile	2022-10-10 09:41:19.000000000 -0700
+++ MuSE-2.0.1/Makefile	2023-05-19 15:57:49.406912000 -0700
@@ -1,17 +1,19 @@
-CPP := g++
-CC := gcc
+CXX := ${CXX} 
+CC := ${CC} 
 LINK := g++
 mkfile_dir := $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
 
 CSOURCES= $(wildcard src/*.c)  
-CPPSOURCES= $(wildcard src/*.cpp)
-OBJS=$(CSOURCES:.c=.c.o)  $(CPPSOURCES:.cpp=.cpp.o)
+CXXSOURCES= $(wildcard src/*.cpp)
+OBJS=$(CSOURCES:.c=.c.o)  $(CXXSOURCES:.cpp=.cpp.o)
 COMMONOBJS= lib/libhts.a lib/libboost_iostreams.a lib/libtcmalloc_minimal.a
 # Warnings is included in WarningsAsErrors to make sure that the warning is enabled.
 Warnings=-Wreturn-type -Warray-bounds -Wmaybe-uninitialized -Waddress
 WarningsAsErrors=$(Warnings) -Werror=return-type -Werror=array-bounds -Werror=address
-CFLAGS=  $(WarningsAsErrors) -Wno-unused-function
-CPPFLAGS=  $(WarningsAsErrors) -Wno-unused-function -std=c++11
+CFLAGS :=  $(WarningsAsErrors) -Wno-unused-function
+LDFLAGS := ${LDFLAGS}
+CXXFLAGS :=  $(WarningsAsErrors) -Wno-unused-function -std=c++11
+CPPFLAGS := ${CPPFLAGS}
 
 RELEASE_FLAGS= -O3 -g
 
@@ -21,26 +23,26 @@
 # Common flags
 COMMONFLAGS += $(INCLUDES)
 
-CXXFLAGS += $(COMMONFLAGS)
-CFLAGS += $(COMMONFLAGS)
 CPPFLAGS += $(COMMONFLAGS)
-COMMONLIBS= -Llib/ -lz -lm -lpthread -lbz2 -lcurl -lcrypto -llzma -fopenmp
+CFLAGS += $(COMMONFLAGS)
+CXXFLAGS += $(COMMONFLAGS)
+COMMONLIBS= -Llib/ -lz -lm -lpthread -lbz2 -lcurl -lcrypto -llzma -fopenmp -lhts -lboost_iostreams 
 
 #LIBS += $(COMMONLIBS) -ltcmalloc
-LIBS += $(COMMONLIBS)
+LIBS := ${LIBS} $(COMMONLIBS)
 
 TARGET = MuSE
-LINKLINE = $(LINK)  -O3 -o $(TARGET) $(OBJS) $(COMMONOBJS) $(MATCHOBJS) $(LIBS)
+LINKLINE = $(LINK)  -O3 -o $(TARGET) $(OBJS) $(LIBS) $(LDFLAGS)
 
 #all:
 all: $(TARGET) 
 .SUFFIXES: .c .cpp .o
 
 %.c.o: %.c
-	$(CC) $(CFLAGS) $(RELEASE_FLAGS) -c $< -o $@
+	$(CC) $(CFLAGS) $(CPPFLAGS) $(RELEASE_FLAGS) -c $< -o $@
 
 %.cpp.o: %.cpp
-	$(CPP) $(CPPFLAGS) $(RELEASE_FLAGS) -c $< -o $@
+	$(CXX) $(CXXFLAGS) $(CPPFLAGS) $(RELEASE_FLAGS) -c $< -o $@
 
 $(TARGET): $(OBJS) Makefile
 	$(LINKLINE)
diff -ruN MuSE-2.0.1.orig/src/index.c MuSE-2.0.1/src/index.c
--- MuSE-2.0.1.orig/src/index.c	2022-10-10 09:41:19.000000000 -0700
+++ MuSE-2.0.1/src/index.c	2023-05-19 10:46:36.715093000 -0700
@@ -4,6 +4,7 @@
 #include "htslib/khash.h"
 #include "htslib/ksort.h"
 #include "htslib/kstring.h"
+#include "htslib/bgzf.h"
 //#include "bam_endian.h"
 // #include "bam.h"
 //#ifdef _USE_KNETFILE
diff -ruN MuSE-2.0.1.orig/src/PBBamFile.cpp MuSE-2.0.1/src/PBBamFile.cpp
--- MuSE-2.0.1.orig/src/PBBamFile.cpp	2022-10-10 09:41:19.000000000 -0700
+++ MuSE-2.0.1/src/PBBamFile.cpp	2023-05-19 17:17:50.819923000 -0700
@@ -7,6 +7,8 @@
 #include "PBBamFile.h"
 #include "htslib/khash.h"
 #include "htslib/bgzf.h"
+#include "htslib/sam.h"
+#include <zlib.h>
 
 
 //READING FILE
